src/perfrecordssh.cpp

```diff
@@ -0,0 +1,215 @@
+    return {};
+}
+
+bool PerfRecordSSH::canTrace(const QString& path)
+{
+    if (m_deviceName.isEmpty())
+        return false;
+
+    // assume best case
+    return true;
+}
+
+bool PerfRecordSSH::canProfileOffCpu()
+{
+    if (m_deviceName.isEmpty())
+        return false;
+    return canTrace(QStringLiteral("events/sched/sched_switch"));
+}
+
+QString perfRecordHelp(const QString& hostname)
+{
+    const QString recordHelp = [hostname]() {
+        static QString help =
+            sshOutput(hostname, {QLatin1String("perf"), QLatin1String("record"), QLatin1String("--help")});
+        if (help.isEmpty()) {
+            // no man page installed, assume the best
+            help = QLatin1String("--sample-cpu --switch-events");
+        }
+        return help;
+    }();
+    return recordHelp;
+}
+
+QString perfBuildOptions(const QString& hostname)
+{
+    const QString buildOptionsHelper = [hostname]() {
+        static QString buildOptions =
+            sshOutput(hostname, {QLatin1String("perf"), QLatin1String("version"), QLatin1String("--build-options")});
+        return buildOptions;
+    }();
+    return buildOptionsHelper;
+}
+
+bool PerfRecordSSH::canSampleCpu()
+{
+    if (m_deviceName.isEmpty())
+        return false;
+    return perfRecordHelp(m_deviceName).contains(QLatin1String("--sample-cpu"));
+}
+
+bool PerfRecordSSH::canSwitchEvents()
+{
+    if (m_deviceName.isEmpty())
+        return false;
+    return perfRecordHelp(m_deviceName).contains(QLatin1String("--switch-events"));
+}
+
+bool PerfRecordSSH::canUseAio()
+{
+    // perf reports "error: Illegal seek" when trying to use aio and streaming data
+    return false;
+}
+
+bool PerfRecordSSH::canCompress()
+{
+    if (m_deviceName.isEmpty())
+        return false;
+    return Zstd_FOUND && perfBuildOptions(m_deviceName).contains(QLatin1String("zstd: [ on  ]"));
+}
+
+bool PerfRecordSSH::isPerfInstalled()
+{
+    if (m_deviceName.isEmpty())
+        return false;
+    return sshExitCode(m_deviceName, {QLatin1String("command"), QLatin1String("-v"), QLatin1String("perf")}) != 0;
+}
+
+void PerfRecordSSH::startRecording(const QStringList& perfOptions, const QString& outputPath,
+                                   const QStringList& recordOptions, const QString& workingDirectory)
+{
+    if (m_recordProcess) {
+        stopRecording();
+    }
+
+    QFileInfo outputFileInfo(outputPath);
+    QString folderPath = outputFileInfo.dir().path();
+    QFileInfo folderInfo(folderPath);
+    if (!folderInfo.exists()) {
+        emit recordingFailed(tr("Folder '%1' does not exist.").arg(folderPath));
+        return;
+    }
+    if (!folderInfo.isDir()) {
+        emit recordingFailed(tr("'%1' is not a folder.").arg(folderPath));
+        return;
+    }
+    if (!folderInfo.isWritable()) {
+        emit recordingFailed(tr("Folder '%1' is not writable.").arg(folderPath));
+        return;
+    }
+
+    qRegisterMetaType<QProcess::ExitStatus>("QProcess::ExitStatus");
```

Comment:
what's this for? I don't see any queued or inter-thread signal/slot emission with this type?

that said, assuming it is required: do you really need to specify the string or is it enough to say `qRegisterMetaType<QProcess::ExitStatus>()`?